notebooks/working/investigating_deconfounder.py

```diff
@@ -532,16 +568,25 @@ def confounder_ppca(X, latent_dim, holdout_portion):
     x_train = np.multiply(1-holdout_mask, X)
     x_vad = np.multiply(holdout_mask, X)
 
-    def ppca_model(data_dim, latent_dim, num_datapoints, stddv_datapoints):
+    def ppca_model(data_dim, latent_dim, num_datapoints, stddv_datapoints, linear = linear):
         w = ed.Normal(loc=tf.zeros([latent_dim, data_dim]),
                     scale=tf.ones([latent_dim, data_dim]),
                     name="w")  # parameter
         z = ed.Normal(loc=tf.zeros([num_datapoints, latent_dim]),
                     scale=tf.ones([num_datapoints, latent_dim]), 
                     name="z")  # local latent variable / substitute confounder
-        x = ed.Normal(loc=tf.multiply(tf.matmul(z, w), 1-holdout_mask),
+        ## linear or quadratic factor model?
+        if linear:
+            x = ed.Normal(loc=tf.multiply(tf.matmul(z, w), 1-holdout_mask),
                     scale=stddv_datapoints * tf.ones([num_datapoints, data_dim]),
                     name="x")  # (modeled) data
+        else: 
+            x = ed.Normal(loc=tf.multiply(tf.matmul(z, w), 1-holdout_mask) 
+                          + tf.multiply(tf.matmul(z**2, w**2), 1-holdout_mask),
+                    scale=stddv_datapoints * tf.ones([num_datapoints, data_dim]),
+                    name="x")
+            
+            
         return x, (w, z)
```

Comment:
Can this go outside of the `confounder_ppca` function? And then just call it directly inside the other function.